pynpoint/processing/psfsubtraction.py

```diff
@@ -189,67 +276,193 @@ def _run_multi_processing(self,
+                res_shape = (len(pca_first), len(wavelength), im_shape[-2], im_shape[-1])
 
-            residuals, res_rot = pca_psf_subtraction(images=star_reshape,
-                                                     angles=parang,
-                                                     pca_number=int(pca_number),
-                                                     pca_sklearn=self.m_pca,
-                                                     im_shape=im_shape,
-                                                     indices=indices)
+            elif self.m_processing_type in ['SDI+ADI', 'ADI+SDI']:
+                res_shape = (len(pca_first), len(pca_secon), len(wavelength),
+                             im_shape[-2], im_shape[-1])
 
-            hist = f'max PC number = {np.amax(self.m_components)}'
+        else:
+            res_shape = (len(pca_first), im_shape[-2], im_shape[-1])
+
+        out_array_mean = np.zeros(res_shape)
+        out_array_medi = np.zeros(res_shape)
+        out_array_weig = np.zeros(res_shape)
+        out_array_clip = np.zeros(res_shape)
+
+        # loop over all different combination of pca_numbers and applying the reductions
+        for i, pca_1 in enumerate(pca_first):
+            for j, pca_2 in enumerate(pca_secon):
+                progress(i+j, len(pca_first)+len(pca_secon), 'Creating residuals...', start_time)
+
+                # process images
+                residuals, res_rot = postprocessor(images=star_reshape,
+                                                   angles=parang,
+                                                   scales=scales,
+                                                   pca_number=(pca_1, pca_2),
+                                                   pca_sklearn=self.m_pca,
+                                                   im_shape=im_shape,
+                                                   indices=indices,
+                                                   processing_type=self.m_processing_type)
+
+                # 1.) derotated residuals
+                if self.m_res_arr_out_ports is not None:
+                    if self.m_processing_type == 'ADI' and not self.m_ifs_data:
+                        self.m_res_arr_out_ports[pca_1].set_all(res_rot)
+                        self.m_res_arr_out_ports[pca_1].copy_attributes(self.m_star_in_port)
+                        self.m_res_arr_out_ports[pca_1].add_history(
+                            'PcaPsfSubtractionModule', f'max PC number = {pca_first}')
 
-            # 1.) derotated residuals
-            if self.m_res_arr_out_ports is not None:
-                self.m_res_arr_out_ports[pca_number].set_all(res_rot)
-                self.m_res_arr_out_ports[pca_number].copy_attributes(self.m_star_in_port)
-                self.m_res_arr_out_ports[pca_number].add_history('PcaPsfSubtractionModule', hist)
+                    else:
+                        if len(pca_first)+len(pca_secon) == 2:
```

Comment:
Remove this if condition. Instead, add an ValueError in the class `__init__` in case of IFS data and there are more than 1 PC used in combination with res_arr_out.

Reply:
I removed it and all other similar if conditions. I also added the requested ValueError.